.github/workflows/clang-tidy.yml

```
@@ -0,0 +1,60 @@
+name: Clang-tidy check
+
+on: [ merge_group, push, pull_request ]
+
+jobs:
+  clang-tidy:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        preset: ["tests-posix-debug", "tests-posix-release", "tests-s32k1xx-debug", "tests-s32k1xx-release"]
```

To include the reference application:
```suggestion
        preset: ["tests-posix-debug", "tests-posix-release", "tests-s32k1xx-debug", "tests-s32k1xx-release", "posix-freertos", "s32k148-freertos-clang"]
```